### PR TITLE
Add Firebolt connector

### PR DESCRIPTION
**Pull request by ptiurin**
## Description
Adding support for [Firebolt](www.firebolt.io) in Trino. Firebolt uses a JDBC driver to connect to the database with SQL syntax close to PostgreSQL. Therefore, this connector was largely based on Postgres.

## Additional context and related issues
Firebolt is a cloud data warehouse. Relevant information on SQL syntax can be found in [docs](https://docs.firebolt.io/).
Notably, [data types](https://docs.firebolt.io/general-reference/data-types.html) and [sql functions](https://docs.firebolt.io/sql-reference/functions-reference/).

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
(X) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-firebolt/pom.xml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    <packaging>trino-plugin</packaging>
+
+    <properties>
+        <air.main.basedir>${project.parent.basedir}</air.main.basedir>
+    </properties>
+
+    <repositories>
+        <repository>
+            <id>repsy</id>
+            <name>Firebolt Private Maven Repository on Repsy</name>
```
Comment (mosabua) [inline plugin/trino-firebolt/pom.xml:22]: I dont think we allow repository adds to a specific plugin. Can you get your artifacts published to Maven Central please?

File: plugin/trino-firebolt/pom.xml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+        <dependency>
+            <groupId>io.airlift</groupId>
+            <artifactId>configuration</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.firebolt</groupId>
+            <artifactId>firebolt-jdbc</artifactId>
+            <version>2.4.5-SNAPSHOT</version>
```
Comment (mosabua) [inline plugin/trino-firebolt/pom.xml:46]: Cant use a snapshot version.

**Review by mosabua — CHANGES_REQUESTED** [review]
Could you also add documentation? 

Had a first glimpse only.

File: plugin/trino-firebolt/pom.xml (hunk 2 above)
Comment (ptiurin) [reply plugin/trino-firebolt/pom.xml:46]: This snapshot contains a couple of fixes to the JDBC driver that's required in this change. A full version of the driver will be released shortly, this is added for the sake of creating this PR and getting some early feedback.
I will update to a full version before this PR is finalised.

File: plugin/trino-firebolt/pom.xml (hunk 1 above)
Comment (ptiurin) [reply plugin/trino-firebolt/pom.xml:22]: I'm looking into publishing on Maven Central, so far I think we should be able to do so without issues. Will update the PR once it's done.

**Comment by ptiurin** [comment]
> Could you also add documentation?
> 

Added documentation in the Connectors section. Let me know if I need to modify any other docs.

File: plugin/trino-firebolt/pom.xml (hunk 2 above)
Comment (mosabua) [reply plugin/trino-firebolt/pom.xml:46]: great

File: plugin/trino-firebolt/pom.xml (hunk 1 above)
Comment (mosabua) [reply plugin/trino-firebolt/pom.xml:22]: thank you

File: docs/src/main/sphinx/connector/firebolt.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
+====================
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/firebolt.rst:1]: Please convert the whole doc to md - you can do so with rst2myst  .. see https://pypi.org/project/rst-to-myst/

File: plugin/trino-firebolt/pom.xml (hunk 2 above)
Comment (ptiurin) [reply plugin/trino-firebolt/pom.xml]: Removed snapshot in favor of `2.4.6`.

File: plugin/trino-firebolt/pom.xml (hunk 1 above)
Comment (ptiurin) [reply plugin/trino-firebolt/pom.xml]: Removed this as we've migrated to Maven Central.

**Comment by ptiurin** [comment]
@trinodb/maintainers Could I get some eyes on this PR please?

**Comment by ebyhr** [comment]
@ptiurin Does Firebolt provide the dockerized image? If not, can your company offer a free account for running CI?

**Comment by ptiurin** [comment]
> @ptiurin Does Firebolt provide the dockerized image? If not, can your company offer a free account for running CI?

There's no dockerised image for Firebolt. I've started a process of provisioning a partner account for you, will get back once that's done. What's the best way of sharing the credentials for it with you?

**Comment by ebyhr** [comment]
Replied in Slack DM.

**Comment by mosabua** [comment]
@ebyhr @ptiurin let me know if I can help somehow

**Comment by ptiurin** [comment]
Sorry for the silence here, I currently have no capacity to work on this change. Feel free to close it if you have to and I will open a new off the latest master when there's interest.

**Pull request closed without merging**